### PR TITLE
Change App Signals Name To Application Signals In Agent Config

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
Change configuration to application_signals from app_signals

# Description of changes
Change configuration to application_signals from app_signals

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Unit tests and starting the operator 
https://github.com/aws/amazon-cloudwatch-agent-operator/pull/144/files
Screenshot 2024-04-08 at 12 43 17 PM

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

**Review by lisguo — DISMISSED** [review]

File: translator/translate/otel/processor/awsappsignals/translator.go [hunk 1]
```diff
@@ -92,10 +96,13 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 	return t.translateCustomRules(conf, configKey, cfg)
 }
 
-func (t *translator) translateMetricLimiterConfig(conf *confmap.Conf, configKey string) (*appsignalsconfig.LimiterConfig, error) {
-	limiterConfigKey := common.ConfigKey(configKey, "limiter")
+func (t *translator) translateMetricLimiterConfig(conf *confmap.Conf, configKey []string) (*appsignalsconfig.LimiterConfig, error) {
```
Comment (bjrara) [inline translator/translate/otel/processor/awsappsignals/translator.go:99]: The configKey is still from `common.AppSignalsConfigKeys[t.dataType]`, can it properly get the configuration from new path?

File: translator/config/schema.json [hunk 2]
```diff
@@ -618,6 +618,90 @@
               },
               "additionalProperties": true
             },
+            "application_signals": {
```
Comment (bjrara) [inline translator/config/schema.json:621]: What if customer configures both `app_signals` and `application_signals` in the configuration? I don't see validation.

**Review by bjrara — COMMENTED** [review]
1. I notice there are `app_signals` option in agent flag, should we rename it as well? 
2. Can we also rename processor name from `awsappsignals` to `awsapplicationsignals`?

File: translator/translate/otel/processor/awsappsignals/translator.go [hunk 3]
```diff
@@ -92,10 +96,13 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
...
+	limiterConfigKey := common.ConfigKey(configKey[0], "limiter")
```
[the remark this answers is not in the document]
Comment (sethAmazon) [reply translator/translate/otel/processor/awsappsignals/translator.go:100]: Not sure what you mean here. This is an array for both application_signals and app_singals to be parsed. 
```
configKey := common.AppSignalsConfigKeys[t.dataType]
```

```
AppSignalsTraces          = ConfigKey(TracesKey, TracesCollectedKey, AppSignals)
	AppSignalsMetrics         = ConfigKey(LogsKey, MetricsCollectedKey, AppSignals)
	AppSignalsTracesFallback  = ConfigKey(TracesKey, TracesCollectedKey, AppSignalsFallback)
	AppSignalsMetricsFallback = ConfigKey(LogsKey, MetricsCollectedKey, AppSignalsFallback)

	AppSignalsConfigKeys = map[component.DataType][]string{
		component.DataTypeTraces:  {AppSignalsTraces, AppSignalsTracesFallback},
		component.DataTypeMetrics: {AppSignalsMetrics, AppSignalsMetricsFallback},
	}
```

File: translator/translate/otel/processor/awsappsignals/translator.go (hunk 1 above)
Comment (sethAmazon) [reply translator/translate/otel/processor/awsappsignals/translator.go:99]: It has both paths [0] is new [1] is old

File: translator/config/schema.json (hunk 2 above)
Comment (sethAmazon) [reply translator/config/schema.json:621]: application_signals should take preference. I'll add a new unit test

**Comment by sethAmazon** [comment]
The flag is internal only but I can make the change.

File: translator/translate/otel/processor/awsappsignals/translator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bjrara) [reply translator/translate/otel/processor/awsappsignals/translator.go:100]: Sorry, didn't capture the structure change.

File: translator/config/schema.json (hunk 2 above)
Comment (sethAmazon) [reply translator/config/schema.json:621]: Added unit test

**Comment by sethAmazon** [comment]
Changed flag

**Review by lisguo — APPROVED** [review]

**Pull request merged**